internal/helm/repository/chart_repository.go

```diff
@@ -308,7 +340,46 @@ func (r *ChartRepository) LoadFromFile(path string) error {
 	if err != nil {
 		return err
 	}
-	return r.LoadIndexFromBytes(b)
+
+	h := sha256.New()
+	if _, err := io.Copy(h, bytes.NewReader(b)); err != nil {
+		return err
+	}
+
+	checksum := fmt.Sprintf("%x", h.Sum(nil))
+	// If optimization is disabled, use YAML index loader and nothing else.
+	if !r.OptimizeIndexLoading {
+		return r.LoadIndex(b, checksum, indexFromYAML)
+	}
+
+	// First check that an optimized index exists, if so load using JSON index loader.
+	// On top of a more efficient unmarshaling process, it will also avoid a sorting
+	// operation across the entire index.
+	jsonIndex := filepath.Join(filepath.Dir(path), fmt.Sprintf("index-%s.json", checksum))
```

Comment:
Does GC removes the json file from disk when a new version is available?

Reply:
Not at the moment. I am still thinking on the best way to approach it given that its name is based on the uniqueness of the index content, meaning that a single `.json` could be shared across multiple `HelmRepository` objects for example.

Reply:
I'm not for sharing anything between repositories, in the future this would be a major blocker to sharding.

Reply:
I'm not for sharing anything between repositories, in the future this would be a major blocker to sharding.

Reply:
Do you mind clarifying how would this be a blocker?
Note this is a resource consumption throttling mechanism and there is no link between the `HelmRepository` objects and the `json` file, apart from the uniqueness of the content it represents. 

The garbage collection of the indexes (`.yaml`) should however differ from the pre-processed (`.json`) ones, based on the non-linear resource consumption impact of the latter.